lib/rspec/core/profiler.rb

```diff
@@ -20,7 +20,9 @@ def example_group_started(notification)
       def example_group_finished(notification)
         return unless notification.group.top_level?
 
-        @example_groups[notification.group][:total_time] =  Time.now - @example_groups[notification.group][:start]
+        if @example_groups[notification.group][:start]
+          @example_groups[notification.group][:total_time] =  Time.now - @example_groups[notification.group][:start]
```

Comment:
`@example_groups[notification.group][:start]` is evaluated twice here and `@example_groups[notification.group]` is evaluated three times, which seems wasteful.  Can we refactor to not re-evaluate things?  Maybe this:

``` ruby
group = @example_groups[notification.group]
group[:total_time] = Time.now - group[:start] if group.key?(:start)
```

Reply:
Sure